llvm/lib/Target/AVR/AVRAsmPrinter.cpp

```diff
@@ -263,11 +263,17 @@ bool AVRAsmPrinter::doFinalization(Module &M) {
     auto *Section = cast<MCSectionELF>(TLOF.SectionForGlobal(&GO, TM));
     if (Section->getName().starts_with(".data"))
       NeedsCopyData = true;
-    else if (Section->getName().starts_with(".rodata") && SubTM->hasLPM())
+    else if (Section->getName().starts_with(".rodata") && SubTM->hasLPM()) {
       // AVRs that have a separate program memory (that's most AVRs) store
-      // .rodata sections in RAM.
-      NeedsCopyData = true;
-    else if (Section->getName().starts_with(".bss"))
+      // .rodata sections in RAM,
+      // but XMEGA3 family maps all flash in the data space.
```

Comment:
Does it actually map the entire flash or just this 32 kb segment?

If it's just this segment that's mapped, is it assumed that we should never copy anything to RAM because the flash will just remap stuff during runtime?

Reply:
Ah, just realized that it wouldn't make sense for a firmware to access more than a single segment anyway, because it wouldn't have enough tam to copy the stuff into anyway (under the current approach where we copy stuff into RAM) :-P